jupyterhub/traitlets.py

```diff
@@ -33,7 +33,7 @@ def validate(self, obj, value):
         return super().validate(obj, value)
 
 
-class ByteSpecification(Integer):
+class ByteSpecification(Float):
```

Comment:
If it should be cast back to int after multiplying, I think this can go back to Integer.